File: CyScat/round2.jl
"""
round2 - Rounding to specified decimal places

Julia translation of MATLAB round2.m
Original MATLAB code by Curtis Jin (jsirius@umich.edu)
"""

"""
    round2(decimal::Int, number)

Round a number to specified decimal places.

# Arguments
- `decimal::Int`: Number of decimal places
- `number`: Number(s) to round (scalar or array)

# Returns
- Rounded value(s)
"""
function round2(decimal::Int, number)
    rounded_value = number * 10^decimal
    rounded_value = trunc.(rounded_value)
    rounded_value = rounded_value * 10^(-decimal)
    return rounded_value
end

File: python/CyScat/round2.py
"""
round2 - Rounding to specified decimal places

Python translation of MATLAB round2.m
"""

import numpy as np


def round2(decimal, number):
    """
    Round a number to specified decimal places.

    Parameters:
    -----------
    decimal : int
        Number of decimal places
    number : float or ndarray
        Number(s) to round

    Returns:
    --------
    rounded_value : float or ndarray
        Rounded value(s)
    """
    rounded_value = number * 10 ** decimal
    rounded_value = np.fix(rounded_value)
    rounded_value = rounded_value * 10 ** (-decimal)

    return rounded_value
